### DISCUSSION TITLE
Is there a way to group vulnerabilities found on multiple URL's?

### DISCUSSION THREAD

**Discussion by osean-man** [projectdiscovery/nuclei#4076] [Q&A]
If I run something like `katana -u https://security-crawl-maze.app | nuclei ...other options` I will end up with lots of duplicates. There are instances where duplicates are wanted, but for example, finding HTTP headers  I will end up with lots of findings for the same thing. If I see a missing header, chances are I'm going to assume it's missing everywhere. For example, I may get 1 finding for every URL Katana finds for [http-missing-security-headers:x-frame-options]. I'm curious if there is a way to either:

1) Have Nuclei just show me the first finding for that type/template or better;
2) In the jsonl output, just show all the URL's it was found.*

*I asked one of the PD guys at Defcon about this and they said they should already be grouped, but I don't observe that behavior. 

I'm using Nuclei 2.9.11.

Thanks!

**Comment by MetzinAround** [comment]
@ehsandeep could you help answer this?

**Comment by ehsandeep** [comment]
@osean-man JSONL output events are generated per match; if you are looking for grouped view of results, you can consider using `-markdown-export` option that let you view the results by host, severity, or template.

See https://github.com/projectdiscovery/nuclei/pull/3961 for example.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
